README.md

```diff
@@ -11,572 +11,528 @@
 
-``` haskell
-cat = S.fold (FH.writeChunks stdout) . S.unfold FH.readChunks
-main = withArg cat
-```
+## More examples
 
-### cp
+Many more examples can be found in the [streamly-examples
+repository](https://github.com/composewell/streamly-examples).
 
-``` haskell
-cp src dst = S.fold (FH.writeChunks dst) $ S.unfold FH.readChunks src
-main = withArg2 cp
-```
+Streamly comes equipped with a very powerful set of abstractions to accomplish
+any kind of programming tasks that you may want to throw at it. It provides,
+streams, arrays, file-io, fsnotify, network-io, time domain programming
+(reactive programming). See the [streamly
+documentation](https://streamly.composewell.com) to know more.
 
-### wc -l
+## Concurrency
 
-``` haskell
-wcl = S.length . S.splitOn (== 10) FL.drain . S.unfold FH.read
-main = withArg wcl >>= print
-```
+Streamly uses lock-free synchronization for concurrent operations. It employs
+auto-scaling of the degree of concurrency based on demand. For CPU bound tasks
+it tries to keep the threads close to the number of CPUs available whereas for
+IO bound tasks more threads can be utilized. Parallelism can be utilized with
+little overhead even if the task size is very small.  See [concurrency
+benchmarks](https://github.com/composewell/concurrency-benchmarks) for detailed
+performance results and a comparison with the `async` package.
 
-### Average Line Length
+## Performance
 
-``` haskell
-avgll =
-      S.fold avg
-    . S.splitOn (== 10) FL.length
-    . S.unfold FH.read
+As you have seen above in the word count example, streamly enables
+highly modular abstractions with the best possible performance (close to
+an equivalent C program).
 
-    where avg      = (/) <$> toDouble FL.sum <*> toDouble FL.length
-          toDouble = fmap (fromIntegral :: Int -> Double)
+For example, none of the existing Haskell libraries can work with byte
+level streams with acceptable performance. Streamly provides excellent
+performance even for byte level stream operations, it is made possible
+by employing efficient abstractions like `Unfold`s and terminating
+`Fold`s. Byte level stream operations make programming simpler because
+you do not have to deal with chunking and re-combining.
 
-main = withArg avgll >>= print
-```
+If you can write a program significantly faster in some other way or
+with some other language, please let us know and we will improve. We
+still have some areas to improve but we will surely get there for most
+cases.
 
-### Line Length Histogram
+## Benchmarks
 
-`classify` is not released yet, and is available in
-`Streamly.Internal.Data.Fold`
+When it comes to stream operations, Haskell lists is the fastest
+existing implementation even though it supports only pure (not monadic)
+streams. Other streaming libraries are many times slower compared to
+lists, so lists is our gold standard for performance comparison.
```

Comment:
> Other streaming libraries are many times slower compared to lists

Don't know how people take direct comparative statements.

Reply:
That's a known and acknowledged fact, also proven by streaming-benchmarks below. We can still remove or rephrase it if it does not sound ok.